### DISCUSSION TITLE
How to create PR with git commands

### DISCUSSION THREAD

**Discussion by Enva21** [community/community#67497] [New to GitHub]
Whats best approach to create PR with git

**Comment by KostaD02** [comment]
Creating a Pull Request and Merging (Using Git Commands)

### Ensure you are on the Correct Branch:

```bash
git checkout target-branch
```

### Pull the Latest Changes:

```bash
git pull origin target-branch
```

### Create a New Branch for Your Feature or Fix:

```bash
git checkout -b my-feature-branch
```

### Make Changes and Commit:

- Make your changes and stage them for commit:
- ```git add .```
- Commit your changes: ```git commit -m "Message"```

### Push Your Branch to the Remote Repository:

```bash 
git push origin my-feature-branch
```

### Create a Pull Request:

Go to your repository on GitHub.
Switch to the new branch you just pushed.
Click on the "Pull Request" button.
Provide a title and description for your pull request.

### Review and Merge the Pull Request:

Wait for reviewers to approve your pull request.
Once approved, click on the "Merge" button.

### Delete the Feature Branch (Optional):

- If you're done with the feature branch, you can delete it locally and on the remote repository:

```bash
git branch -d my-feature-branch
git push origin --delete my-feature-branch
```

### Update Your Local Repository:
- Switch back to the target branch: ```git checkout target-branch```
- Pull the latest changes: ```git pull origin target-branch```

**Reply by Enva21** [reply]
Thanks

**Comment by queenofcorgis** [comment]
GitHub's [Acceptable Use Policies](https://docs.github.com/en/site-policy/acceptable-use-policies/github-acceptable-use-policies) prohibits coordinated or inauthentic activity like rapid questions and answers. As a result, we'll be unmarking the answer and locking this post. Please note any future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
